types/index.d.ts

```diff
@@ -126,26 +167,149 @@ export namespace Service {
     noWait: boolean;
   }
 
-  interface Server extends EventEmitter {
+  class Server extends EventEmitter {
+    constructor(svc: any, opts: any);
+
     readonly service: Service;
     // on<message>()
+
     activeChannels(): ServerChannel[];
     createChannel(transport: Transport, options?: Partial<ChannelCreateOptions>): ServerChannel;
     onMessage<T>(name: string, handler: (arg1: any, callback: Callback<T>) => void): this;
     remoteProtocols(): Protocol[];
     use(...args: any[]): this;
   }
 
-  interface Client extends EventEmitter {
+  class Client extends EventEmitter {
+    constructor(svc: any, opts: any);
     activeChannels(): ClientChannel[];
     createChannel(transport: Transport, options?: Partial<ChannelCreateOptions>): ClientChannel;
     destroyChannels(options?: Partial<ChannelDestroyOptions>): void;
     emitMessage<T>(name: string, req: any, options?: any, callback?: Callback<T>): void // TODO
     remoteProtocols(): Protocol[];
     use(...args: any[]): this;
   }
+}
+
+export namespace streams {
+  class BlockDecoder {
+    constructor(opts?: any);
+    static defaultCodecs(): any;
+  }
+
+  class BlockEncoder {
+    constructor(schema: Schema, opts: any);
+    static defaultCodecs(): any;
+  }
+
+  class RawDecoder {
+    constructor(schema: Schema, opts: any);
+  }
+
+  class RawEncoder {
+    constructor(schema: Schema, opts: any);
+  }
+}
+
+export namespace types {
+  class ArrayType extends Type {
+    constructor(schema: Schema, opts: any);
+    readonly itemsType: Type;
+    random(): ArrayType;
+  }
+
+  class BooleanType extends Type {  // TODO: Document this on the wiki
```

Comment:
Ah, you're right; these aren't in the Wiki. The reason they were not included originally is they don't have any custom methods (other than those in the `Type` superclass). I'll add them when I get a chance.